packages/react-native-tracer-provider/src/EmbraceNativeTracerProvider.ts

```diff
@@ -0,0 +1,143 @@
+import {NativeModules, AppState, Platform} from "react-native";
+import {useEffect, useState} from "react";
+import {
+  context,
+  ContextManager,
+  Tracer,
+  TracerProvider,
+} from "@opentelemetry/api";
+
+import {logWarning} from "./util";
+import {
+  EmbraceNativeTracerProviderConfig,
+  SpanContextSyncBehaviour,
+  UseEmbraceNativeTracerProviderResult,
+} from "./types";
+import {TracerProviderModule} from "./TracerProviderModule";
+import {StackContextManager} from "./StackContextManager";
+import {EmbraceNativeTracer} from "./EmbraceNativeTracer";
+
+/**
+ * useEmbraceNativeTracerProvider makes sure that the Embrace SDK has been installed and started and
+ * then sets up a EmbraceNativeTracerProvider
+ *
+ * The EmbraceNativeTracerProvider implements a TracerProvider over the native Embrace Android and iOS SDKs.
+ * Thin wrapped objects representing Tracers and Spans are maintained at the JS level and use Native Modules to
+ * call down to the SDKs to perform the actual operations on them.
+ *
+ * The JS side of its implementation is modelled after [opentelemetry-sdk-trace-base](https://github.com/open-telemetry/opentelemetry-js/tree/main/packages/opentelemetry-sdk-trace-base)
+ */
+export const useEmbraceNativeTracerProvider = (
```

Comment:
can we move the hook into a different place? as per the name of this file (`EmbraceNativeTracerProvider`) I though I was going to see only a class but it is also containing the hook. I would suggest to move the hook to its own place `useEmbraceNativeTracerProvider`

Reply:
makes sense, moved to its own `useEmbraceNativeTracerProvider.ts` file